libraries/clickhouse-helpers/clickhouse_helpers/migrations/migration_files/20220217_00_add_epoch_table.sql

```diff
@@ -0,0 +1,18 @@
+CREATE TABLE vital_data_epoch
+(
+    customer INT,
+    type smallint,
+    valueType Int8,
+    doubleValue double precision NULL,
+    longValue INT NULL,
+    booleanValue INT NULL,
+    timezoneOffset smallint,
+    startTimestamp bigint,
+    endTimestamp bigint,
+    createdAt bigint,
+    source SMALLINT
+)
+ENGINE = ReplacingMergeTree()
```

Comment:
There are is no `UPSERT` or `ON CONFLICT` in clickhouse. Workaround is to use the `ReplacingMergeTree` table engine, which periodically clears out duplicates in the background. This cleanup can also be triggered manually with an `OPTIMIZE` query on the table (which is what I use as a workaround.)

Reply:
To be notified if cleaning duplicates fails for any reason, we should consider enabling [this setting](https://clickhouse.com/docs/en/operations/settings/settings/#setting-optimize_throw_if_noop) to throw a warning on failure.

Reply:
created a task an issue and assigned myself